azure-pipelines/templates/steps/nightly-snapshot-releases/common.yml

```diff
@@ -0,0 +1,71 @@
+# File: azure-pipelines\templates\steps\nightly-snapshot-releases\common.yml
+# Description: Template to assemble and publish different versions of common
+# Variable: 'ENV_VSTS_MVN_ANDROIDCOMMON_USERNAME' was defined in the Variables tab
+# Variable: 'mvnAccessToken' was defined in the Variables tab
+# https://docs.microsoft.com/en-us/azure/devops/organizations/accounts/use-personal-access-tokens-to-authenticate
+# Variable: 'versionType' type of version (Latest, Snapshot)
+
+parameters:
+- name: versionType
+  default: Latest
+  values:
+  - Snapshot
+  - Latest
+
+jobs:
+- job: Phase_1
+  displayName: Phase 1
+  timeoutInMinutes: 90
+  cancelTimeoutInMinutes: 1
+  pool:
+    name: Hosted VS2017
+  steps:
+  - checkout: self
+    clean: true
+    submodules: recursive
+    persistCredentials: True
+  - template: ../credscan-policheck.yml
+    parameters:
+      policheckCmdLineArgsDir: common
+  - task: CmdLine@1
+    displayName: Set MVN Access Token in Environment
+    inputs:
+      filename: echo
+      arguments: '##vso[task.setvariable variable=ENV_VSTS_MVN_ANDROIDCOMMON_ACCESSTOKEN]$(mvnAccessToken)'
+  - task: Gradle@2
+    displayName: ${{ parameters.versionType }} version update
+    inputs:
+      tasks: common:version${{ parameters.versionType }}
+      publishJUnitResults: false
+  - task: Gradle@1
+    name: Gradle1
+    displayName: Assemble Release
+    inputs:
+      tasks: 'clean common:build common:assembleRelease'
+      publishJUnitResults: false
+      jdkArchitecture: x86
+      sqAnalysisBreakBuildIfQualityGateFailed: false
+  - task: Gradle@2
+    displayName: Run unit tests
+    inputs:
+      tasks: testDebugUnitTest
```

Comment:
>testDebugUnitTest [](http://example.com/codeflow?start=13&length=17)

We should be building and using **release** variants here.
As that is what we will eventually ship to customers.

Reply:
@shahzaibj also tagging you here for your opinion because this is a pipeline previously created by you I think, Thanks

Reply:
The test may fail. We will have to run them first and find out. This is because in the past we may have written tests that may have taken a hard dependency on the build type being `debug` which is usually the default build type selected in the developer's IDE. This would primarily be around tests that are potentially testing broker validation as the behavior here used to different between debug and release. However, this has probably now gone way with my changes in https://github.com/AzureAD/microsoft-authentication-library-common-for-android/pull/1191 and we can probably now just do `BrokerValidator.setShouldTrustDebugBrokers(true)` in the tests and that should cover both debug and release scenarios.

I am personally fine either way (testing debug vs release - given that release tests are passing) and if we want to move to testing release that's great but it doesn't necessarily have to be done as part of this PR. (I would let you make the decision)

Reply:
I would prefer to do testing on release variants, just for the fact that those are the bits customers will use. 
You can create a separate workitem if it turn out to be a bigger task

Reply:
Thanks folks, I would keep your comments in mind and create a Backlog Item to revisit this once I ram up with the test case and finish the setup of more branches 

 https://identitydivision.visualstudio.com/Engineering/_workitems/edit/1386380
OK?

Reply:
That sounds good to me